security/pkg/pki/util/generate_cert.go

```diff
@@ -81,19 +82,39 @@ type CertOptions struct {
 
 	// If true, the private key is encoded with PKCS#8.
 	PKCS8Key bool
+
+	// Whether this certificate uses Elliptical Curve. Mutually
+	// exclusive with respect to RSAKeySize.
+	IsEC bool
 }
 
 // GenCertKeyFromOptions generates a X.509 certificate and a private key with the given options.
 func GenCertKeyFromOptions(options CertOptions) (pemCert []byte, pemKey []byte, err error) {
-	// Generate a RSA private&public key pair.
+	if options.IsEC && options.RSAKeySize > 0 {
+		return nil, nil, fmt.Errorf("cert generation fails due to both RSA key generation and EC generation are specified")
+	}
+
+	// Generate the appropriate private&public key pair based on options.
 	// The public key will be bound to the certificate generated below. The
 	// private key will be used to sign this certificate in the self-signed
 	// case, otherwise the certificate is signed by the signer private key
 	// as specified in the CertOptions.
-	priv, err := rsa.GenerateKey(rand.Reader, options.RSAKeySize)
+	if options.IsEC {
+		ecPriv, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
```

Comment:
P-256 is plenty secure.

Reply:
But Wireguard is using 25519 - and it would be very valuable to integrate nicely ( including provisioning of keys ). 

Why not add a clear pair of options - one for the default root CA to use, and one for workloads.

In many cases the root may be RSA, but you can still use EC for workloads.

Reply:
I think without providing all variants on the get go, we wanted to start simple saying if you provide istiod ECC key/cert all generated workload key pairs are also  ECC. If you provide RSA or don’t give a root everything is RSA.

Let’s keep it simple without supporting everything. We can always add all variations incrementally.